### PR TITLE
Parameterize frame IDs

### LINKED ISSUE #1726: Use Parameterized frame_id
#### Feature description
Use parameterized frame_id.

#### Implementation considerations
Modification is needed for
- https://github.com/ros-planning/navigation2/blob/13633cd3efdaf2a206bb9df0f76e4f5e8e996d2b/nav2_recoveries/src/recovery_server.cpp#L75
  - Using odom and base_link
- https://github.com/ros-planning/navigation2/blob/13633cd3efdaf2a206bb9df0f76e4f5e8e996d2b/nav2_recoveries/plugins/back_up.cpp#L56-L70
  - Using odom and base_link
- https://github.com/ros-planning/navigation2/blob/13633cd3efdaf2a206bb9df0f76e4f5e8e996d2b/nav2_recoveries/plugins/spin.cpp#L73-L94
  - Using odom and base_link
- https://github.com/ros-planning/navigation2/blob/13633cd3efdaf2a206bb9df0f76e4f5e8e996d2b/nav2_costmap_2d/src/collision_checker.cpp#L176
  - Using map and base_link
  - I think that costmap_.getRobotPose should be used.
- https://github.com/ros-planning/navigation2/blob/13633cd3efdaf2a206bb9df0f76e4f5e8e996d2b/nav2_behavior_tree/plugins/condition/goal_reached_condition.cpp#L73
  - Using map and base_link

### PR DESCRIPTION
**Pull request by naiveHobo**
---

## Basic Info

Original PR:  #1729

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | closes  #1726 |
| Primary OS tested on | Ubuntu 18.04 |

### CODE REVIEW

**Comment by SteveMacenski** [comment]
the builds are failing for some really odd reason

**Comment by SteveMacenski** [comment]
I'll wait to review on #1735 which will inevitably cause this to have merge conflicts

**Comment by ruffsl** [comment]
Comparing the `checksum.txt` files between the two jobs, it looks like the test job pulled a newer image from DockerHub as compared to what the build job used. See the dates of the ros_entrypoint file:

* build
  * https://11424-135363400-gh.circle-artifacts.com/0/opt/underlay_ws/checksum.txt
* test
  * https://11429-135363400-gh.circle-artifacts.com/0/opt/underlay_ws/checksum.txt

Looks like an unlucky event where DockerHub finished building and pushed/tagged a new image after the CI build job started, but before the test job started. I've asked CircleCI for a means of getting and setting the docker image sha digest, such that all downstream jobs in a workflow pull the same image.

https://discuss.circleci.com/t/introspect-image-info-from-inside-docker-executor/31620

This may be something that github actions could do better at, but alas more future work.

**Comment by SteveMacenski** [comment]
@ruffsl so is this a temporary state we just need to wait for dockerhub to turn over or is there something active I can do?

**Comment by ruffsl** [comment]
As soon as you see it you can just rerun the workflow, and it'll all sort itself out if the build job reruns and creates a new cash for the newer image for the next test job to restore from. 

I can follow up with the few changes to reduce the likelihood of it happening given some trade-offs.

**Comment by SteveMacenski** [comment]
@ruffsl which workflow are you talking about? The CircleCI one? I've reset that to run a few times and still didn't work - so perhaps you mean dockerhub?

**Comment by ruffsl** [comment]
The tests are working and running, they're just failing for this PR now because there's some issue that's causing the tests themselves to fail. Check the test results for the failing tests.

**Comment by SteveMacenski** [comment]
And these builds for other PRs pending? 

https://circleci.com/gh/ros-planning/navigation2/11440?utm_campaign=vcs-integration-link&
utm_medium=referral&utm_source=github-build-link

https://circleci.com/gh/ros-planning/navigation2/11447?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: nav2_bringup/bringup/params/nav2_params.yaml [hunk 1]
```diff
@@ -243,6 +243,21 @@ planner_server_rclcpp_node:
...
+recoveries_server:
+  ros__parameters:
+    costmap_topic: local_costmap/costmap_raw
+    footprint_topic: local_costmap/published_footprint
+    cycle_frequency: 10.0
+    plugin_names: ["spin", "back_up", "wait"]
+    plugin_types: ["nav2_recoveries/Spin", "nav2_recoveries/BackUp", "nav2_recoveries/Wait"]
+    global_frame: odom
+    robot_base_frame: base_link
+    use_sim_time: true
```
Comment (ymd-stella) [inline nav2_bringup/bringup/params/nav2_params.yaml:255]: Do we also need to make changes to the following code to set this parameter?
https://github.com/ros-planning/navigation2/blob/9aafde139dffc3d69cfa6837145e0cca74f1ede7/nav2_bringup/bringup/launch/navigation_launch.py#L117

Comment (naiveHobo) [reply nav2_bringup/bringup/params/nav2_params.yaml:255]: Ah that's right, I missed this.

**Comment by ruffsl** [comment]
Looks like it might be an upstream issue from the nightly archive used:

```
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error at /opt/ros/foxy/share/connext_cmake_module/cmake/Modules/FindConnext.cmake:326 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  /opt/ros/foxy/share/rosidl_typesupport_connext_c/cmake/rosidl_typesupport_connext_c-extras.cmake:5 (find_package)
```

Not sure why it'd only block the release build and not the debug one too.

**Comment by SteveMacenski** [comment]
@naiveHobo please fix merge conflicts and add `transform_timeout: 0.1` to the yaml files for recovery, bt navigator

**Comment by SteveMacenski** [comment]
@naiveHobo ping me when CI passes and I'll merge

**Comment by naiveHobo** [comment]
@SteveMacenski good to go?

**Pull request merged**